tests/components/esphome/test_entity.py

```diff
@@ -216,6 +216,7 @@ async def test_entities_removed_after_reload(
 
     assert await hass.config_entries.async_setup(entry.entry_id)
     await hass.async_block_till_done()
+    await hass.async_block_till_done()
```

Comment:
Why do we call here `hass.async_block_till_done()` twice?
I'm interested when we need a second one :)

Reply:
Usually its because the event loop is taking less time and the test wasn't actually relying on a hass task and instead relying on things being slower and is now running too soon because there is less overhead.

Usually go back and do another pass (like #110867) as adding another block isn't a great solution long term.

Reply:
I think for this case we might be able to watch for state change events instead and continue on when we get the one we want (or timeout)

Reply:
https://github.com/home-assistant/core/pull/111179